### PR TITLE
refactor CoordsXY(ZD) 

### PR DESCRIPTION
**Pull request by 0Cufox0**
#9495

### CODE REVIEW

**Comment by 0Cufox0** [comment]
I think this is done

**Comment by janisozaur** [comment]
> I think this is done

That's an odd thing to say when all the CI runs are red.

**Comment by 0Cufox0** [comment]
> > I think this is done
> 
> That's an odd thing to say when all the CI runs are red.

ya it got me a bit confused, I looked into it I think the things I did in LargeSceneryRemoveAction caused it(a coordinate is out of range somewhere) I looked into it and I can't find anything as the values from before were big coordinates so I didn't have to do anything, I admit this needs more time I'll take a look at everything in a bit.

**Comment by 0Cufox0** [comment]
the last commit fixed something, but whenever I start up the issue8203 replay manually I get an ingame error message that says that I can't place an item.

**Comment by 0Cufox0** [comment]
I did some experimenting, so one of the CI tests runs a replay that overflows the Z value in a mazeSetTrackAction object, before it worked well because the position was an unsigned int so when it got overflown it just got down to 0, after refactoring the z position is a signed int 32 bit int so it goes down into -2^31, apparently feeding a negative value to the stream in the Serialise function crashes the game which explains why the CI test was failing....
I dealt with it by casting the 32 bit int into an unsigned 16 bit int(the way it was before) and it works just fine... I need more advice on this cause the game still shows an error that says an object can't placed(remember it's -2^31)...

File: src/openrct2/actions/MazeSetTrackAction.hpp [hunk 1]
```diff
@@ -69,10 +69,8 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
 
     void Serialise(DataSerialiser & stream) override
     {
-        GameAction::Serialise(stream);
-        uint16_t _x = _loc.x , _y = _loc.y , _z = _loc.z;
-        Direction _direction = _loc.direction;   
-        stream << DS_TAG(_x) << DS_TAG(_y) << DS_TAG(_z) << DS_TAG(_direction) << DS_TAG(_initialPlacement) << DS_TAG(_rideIndex) << DS_TAG(_mode);
+        GameAction::Serialise(stream);   
+        stream << DS_TAG(_loc.x) << DS_TAG(_loc.y) << DS_TAG(_loc.z) << DS_TAG(_loc.direction) << DS_TAG(_initialPlacement) << DS_TAG(_rideIndex) << DS_TAG(_mode);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/MazeSetTrackAction.hpp]: You can just pass loc as an individual item there should already be a serialisation for it

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 2]
```diff
@@ -485,7 +485,8 @@ int32_t viewport_interaction_right_click(int32_t x, int32_t y)
 static void viewport_interaction_remove_scenery(TileElement* tileElement, int32_t x, int32_t y)
 {
     auto removeSceneryAction = SmallSceneryRemoveAction(
-        x, y, tileElement->base_height, tileElement->AsSmallScenery()->GetSceneryQuadrant(),
+        { x, y, tileElement->base_height << 3 },
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp]: use * 8 for this. In general we are trying to move away from << >>. There are certain situations that this can't be done (some rendering functions) but this is not one of them.

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 3]
```diff
@@ -563,7 +564,7 @@ static void viewport_interaction_remove_park_wall(TileElement* tileElement, int3
     }
     else
     {
-        TileCoordsXYZD wallLocation = { x >> 5, y >> 5, tileElement->base_height, tileElement->GetDirection() };
+        CoordsXYZD wallLocation = { x, y, tileElement->base_height << 3, tileElement->GetDirection() };
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp]: as above

File: src/openrct2-ui/interface/ViewportInteraction.cpp [hunk 4]
```diff
@@ -584,8 +585,7 @@ static void viewport_interaction_remove_large_scenery(TileElement* tileElement,
     }
     else
     {
-        auto removeSceneryAction = LargeSceneryRemoveAction(
-            x, y, tileElement->base_height, tileElement->GetDirection(), tileElement->AsLargeScenery()->GetSequenceIndex());
+        auto removeSceneryAction = LargeSceneryRemoveAction({ x, y, tileElement->base_height << 3, tileElement->GetDirection() }, tileElement->AsLargeScenery()->GetSequenceIndex());
```
Comment (duncanspumpkin) [inline src/openrct2-ui/interface/ViewportInteraction.cpp]: as above

File: src/openrct2-ui/windows/Sign.cpp [hunk 5]
```diff
@@ -232,11 +232,8 @@ static void window_sign_mouseup(rct_window* w, rct_widgetindex widgetIndex)
                 tile_element++;
             }
 
-            auto sceneryRemoveAction = LargeSceneryRemoveAction(
-                x, y, tile_element->base_height, tile_element->GetDirection(),
-                tile_element->AsLargeScenery()->GetSequenceIndex());
+            auto sceneryRemoveAction = LargeSceneryRemoveAction({ x, y, tile_element->base_height << 3, tile_element->GetDirection() }, tile_element->AsLargeScenery()->GetSequenceIndex());
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Sign.cpp]: as above.

File: src/openrct2-ui/windows/Sign.cpp [hunk 6]
```diff
@@ -488,7 +485,7 @@ static void window_sign_small_mouseup(rct_window* w, rct_widgetindex widgetIndex
                 }
                 tile_element++;
             }
-            TileCoordsXYZD wallLocation = { x >> 5, y >> 5, tile_element->base_height, tile_element->GetDirection() };
+            CoordsXYZD wallLocation = { x , y , tile_element->base_height << 3, tile_element->GetDirection() };
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/Sign.cpp]: as above

File: src/openrct2/actions/ClearAction.hpp [hunk 7]
```diff
@@ -152,7 +152,7 @@ DEFINE_GAME_ACTION(ClearAction, GAME_COMMAND_CLEAR_SCENERY, GameActionResult)
                     case TILE_ELEMENT_TYPE_PATH:
                         if (_itemsToClear & CLEARABLE_ITEMS::SCENERY_FOOTPATH)
                         {
-                            auto footpathRemoveAction = FootpathRemoveAction(x * 32, y * 32, tileElement->base_height);
+                            auto footpathRemoveAction = FootpathRemoveAction({ x << 5, y << 5, tileElement->base_height << 3 });
```
Comment (duncanspumpkin) [inline src/openrct2/actions/ClearAction.hpp]: As above replace << 5 with *32 and << 3 with *8.

File: src/openrct2/actions/ClearAction.hpp [hunk 8]
```diff
@@ -169,7 +169,8 @@ DEFINE_GAME_ACTION(ClearAction, GAME_COMMAND_CLEAR_SCENERY, GameActionResult)
                         if (_itemsToClear & CLEARABLE_ITEMS::SCENERY_SMALL)
                         {
                             auto removeSceneryAction = SmallSceneryRemoveAction(
-                                x * 32, y * 32, tileElement->base_height, tileElement->AsSmallScenery()->GetSceneryQuadrant(),
+                                { x << 5, y << 5, tileElement->base_height << 3 },
```
Comment (duncanspumpkin) [inline src/openrct2/actions/ClearAction.hpp]: as above

File: src/openrct2/actions/ClearAction.hpp [hunk 9]
```diff
@@ -186,7 +187,8 @@ DEFINE_GAME_ACTION(ClearAction, GAME_COMMAND_CLEAR_SCENERY, GameActionResult)
                     case TILE_ELEMENT_TYPE_WALL:
                         if (_itemsToClear & CLEARABLE_ITEMS::SCENERY_SMALL)
                         {
-                            TileCoordsXYZD wallLocation = { x, y, tileElement->base_height, tileElement->GetDirection() };
+                            CoordsXYZD wallLocation = { x << 5, y << 5, tileElement->base_height << 3,
```
Comment (duncanspumpkin) [inline src/openrct2/actions/ClearAction.hpp]: as above.

File: src/openrct2/actions/ClearAction.hpp [hunk 10]
```diff
@@ -204,7 +206,7 @@ DEFINE_GAME_ACTION(ClearAction, GAME_COMMAND_CLEAR_SCENERY, GameActionResult)
                         if (_itemsToClear & CLEARABLE_ITEMS::SCENERY_LARGE)
                         {
                             auto removeSceneryAction = LargeSceneryRemoveAction(
-                                x * 32, y * 32, tileElement->base_height, tileElement->GetDirection(),
+                                { x << 5, y << 5, tileElement->base_height << 3, tileElement->GetDirection() },
```
Comment (duncanspumpkin) [inline src/openrct2/actions/ClearAction.hpp]: as above

File: src/openrct2/actions/FootpathRemoveAction.hpp [hunk 11]
```diff
@@ -115,7 +111,7 @@ DEFINE_GAME_ACTION(FootpathRemoveAction, GAME_COMMAND_REMOVE_PATH, GameActionRes
     {
         bool getGhostPath = GetFlags() & GAME_COMMAND_FLAG_GHOST;
 
-        TileElement* tileElement = map_get_footpath_element(_x / 32, _y / 32, _z);
+        TileElement* tileElement = map_get_footpath_element(_loc.x >> 5, _loc.y >> 5, _loc.z >> 3);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/FootpathRemoveAction.hpp]: /32, /8

File: src/openrct2/actions/FootpathRemoveAction.hpp [hunk 12]
```diff
@@ -161,7 +157,7 @@ DEFINE_GAME_ACTION(FootpathRemoveAction, GAME_COMMAND_REMOVE_PATH, GameActionRes
                 continue;
 
             auto bannerRemoveAction = BannerRemoveAction(
-                { x, y, tileElement->base_height * 8, tileElement->AsBanner()->GetPosition() });
+                { x, y, tileElement->base_height << 3, tileElement->AsBanner()->GetPosition() });
```
Comment (duncanspumpkin) [inline src/openrct2/actions/FootpathRemoveAction.hpp]: *8

File: src/openrct2/actions/GameActionCompat.cpp [hunk 13]
```diff
@@ -23,7 +23,7 @@
 #pragma region PlaceParkEntranceAction
 money32 place_park_entrance(int16_t x, int16_t y, int16_t z, uint8_t direction)
 {
-    auto gameAction = PlaceParkEntranceAction(x, y, z, direction);
+    auto gameAction = PlaceParkEntranceAction({ x, y, z << 4, direction });
```
Comment (duncanspumpkin) [inline src/openrct2/actions/GameActionCompat.cpp]: *16

File: src/openrct2/actions/GameActionCompat.cpp [hunk 14]
```diff
@@ -43,7 +43,7 @@ money32 park_entrance_place_ghost(int32_t x, int32_t y, int32_t z, int32_t direc
 {
     park_entrance_remove_ghost();
 
-    auto gameAction = PlaceParkEntranceAction(x, y, z, direction);
+    auto gameAction = PlaceParkEntranceAction({ x, y, z << 4, (Direction)direction });
```
Comment (duncanspumpkin) [inline src/openrct2/actions/GameActionCompat.cpp]: *16

File: src/openrct2/actions/LargeSceneryRemoveAction.hpp [hunk 15]
```diff
@@ -181,14 +173,14 @@ DEFINE_GAME_ACTION(LargeSceneryRemoveAction, GAME_COMMAND_REMOVE_LARGE_SCENERY,
                 }
             }
 
-            TileElement* sceneryElement = map_get_first_element_at(currentTile.x / 32, currentTile.y / 32);
+            TileElement* sceneryElement = map_get_first_element_at(currentTile.x >> 5, currentTile.y >> 5);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/LargeSceneryRemoveAction.hpp]: /32

File: src/openrct2/actions/LargeSceneryRemoveAction.hpp [hunk 16]
```diff
@@ -222,7 +214,7 @@ DEFINE_GAME_ACTION(LargeSceneryRemoveAction, GAME_COMMAND_REMOVE_LARGE_SCENERY,
 private:
     TileElement* FindLargeSceneryElement() const
     {
-        TileElement* tileElement = map_get_first_element_at(_x / 32, _y / 32);
+        TileElement* tileElement = map_get_first_element_at(_loc.x >> 5, _loc.y >> 5);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/LargeSceneryRemoveAction.hpp]: /32

File: src/openrct2/actions/LargeSceneryRemoveAction.hpp [hunk 17]
```diff
@@ -231,13 +223,13 @@ DEFINE_GAME_ACTION(LargeSceneryRemoveAction, GAME_COMMAND_REMOVE_LARGE_SCENERY,
             if (tileElement->GetType() != TILE_ELEMENT_TYPE_LARGE_SCENERY)
                 continue;
 
-            if (tileElement->base_height != _baseHeight)
+            if (tileElement->base_height != _loc.z >> 3)
```
Comment (duncanspumpkin) [inline src/openrct2/actions/LargeSceneryRemoveAction.hpp]: /8

File: src/openrct2/actions/MazeSetTrackAction.hpp [hunk 18]
```diff
@@ -64,66 +59,59 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
...
-        if (!map_is_location_owned(floor2(_x, 32), floor2(_y, 32), _z) && !gCheatsSandboxMode)
+        if (!map_is_location_owned(floor2(_loc.x, 32), floor2(_loc.y, 32), _loc.z) && !gCheatsSandboxMode)
         {
             res->Error = GA_ERROR::NOT_OWNED;
             res->ErrorMessage = STR_LAND_NOT_OWNED_BY_PARK;
             return res;
         }
 
-        TileElement* tileElement = map_get_surface_element_at(_x / 32, _y / 32);
+        TileElement* tileElement = map_get_surface_element_at(_loc.x >> 5, _loc.y >> 5);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/MazeSetTrackAction.hpp]: /32

File: src/openrct2/actions/MazeSetTrackAction.hpp [hunk 19]
```diff
@@ -64,66 +59,59 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
...
         if (tileElement == nullptr)
         {
             res->Error = GA_ERROR::UNKNOWN;
             res->ErrorMessage = STR_INVALID_SELECTION_OF_OBJECTS;
             return res;
         }
 
-        uint8_t baseHeight = _z >> 3;
-        uint8_t clearanceHeight = (_z + 32) >> 3;
+        uint8_t baseHeight = _loc.z >> 3;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/MazeSetTrackAction.hpp]: *8 and below

File: src/openrct2/actions/MazeSetTrackAction.hpp [hunk 20]
```diff
@@ -204,14 +192,14 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
...
         {
-            footpath_remove_litter(_x, _y, _z);
-            wall_remove_at(floor2(_x, 32), floor2(_y, 32), _z, _z + 32);
+            footpath_remove_litter(_loc.x, _loc.y, _loc.z);
+            wall_remove_at(floor2(_loc.x, 32), floor2(_loc.y, 32), _loc.z, _loc.z + 32);
         }
 
-        uint8_t baseHeight = _z >> 3;
-        uint8_t clearanceHeight = (_z + 32) >> 3;
+        uint8_t baseHeight = _loc.z >> 3;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/MazeSetTrackAction.hpp]: *8

File: src/openrct2/actions/MazeSetTrackAction.hpp [hunk 21]
```diff
@@ -220,10 +208,10 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
             money32 price = (((RideTrackCosts[ride->type].track_price * TrackPricing[TRACK_ELEM_MAZE]) >> 16));
             res->Cost = price / 2 * 10;
 
-            uint16_t flooredX = floor2(_x, 32);
-            uint16_t flooredY = floor2(_y, 32);
+            uint16_t flooredX = floor2(_loc.x, 32);
+            uint16_t flooredY = floor2(_loc.y, 32);
 
-            tileElement = tile_element_insert(_x / 32, _y / 32, baseHeight, 0xF);
+            tileElement = tile_element_insert(_loc.x >> 5, _loc.y >> 5, baseHeight, 0xF);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/MazeSetTrackAction.hpp]: *32

File: src/openrct2/actions/MazeSetTrackAction.hpp [hunk 22]
```diff
@@ -340,7 +328,7 @@ DEFINE_GAME_ACTION(MazeSetTrackAction, GAME_COMMAND_SET_MAZE_TRACK, GameActionRe
                 break;
         }
 
-        map_invalidate_tile(floor2(_x, 32), floor2(_y, 32), tileElement->base_height * 8, tileElement->clearance_height * 8);
+        map_invalidate_tile(floor2(_loc.x, 32), floor2(_loc.y, 32), tileElement->base_height << 3, tileElement->clearance_height << 3);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/MazeSetTrackAction.hpp]: *8

File: src/openrct2/actions/PlaceParkEntranceAction.hpp [hunk 23]
```diff
@@ -83,20 +77,20 @@ DEFINE_GAME_ACTION(PlaceParkEntranceAction, GAME_COMMAND_PLACE_PARK_ENTRANCE, Ga
                 GA_ERROR::INVALID_PARAMETERS, STR_CANT_BUILD_PARK_ENTRANCE_HERE, STR_ERR_TOO_MANY_PARK_ENTRANCES);
         }
 
-        int8_t zLow = _z * 2;
+        int8_t zLow = _loc.z >> 3;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/PlaceParkEntranceAction.hpp]: /8

File: src/openrct2/actions/PlaceParkEntranceAction.hpp [hunk 24]
```diff
@@ -121,31 +115,29 @@ DEFINE_GAME_ACTION(PlaceParkEntranceAction, GAME_COMMAND_PLACE_PARK_ENTRANCE, Ga
...
-        parkEntrance.x = _x;
-        parkEntrance.y = _y;
-        parkEntrance.z = _z * 16;
-        parkEntrance.direction = _direction;
+        parkEntrance = _loc;
+    
         gParkEntrances.push_back(parkEntrance);
 
-        int8_t zLow = _z * 2;
+        int8_t zLow = _loc.z >> 3;
```
Comment (duncanspumpkin) [inline src/openrct2/actions/PlaceParkEntranceAction.hpp]: /8

File: src/openrct2/actions/SmallSceneryRemoveAction.hpp [hunk 25]
```diff
@@ -143,7 +135,7 @@ DEFINE_GAME_ACTION(SmallSceneryRemoveAction, GAME_COMMAND_REMOVE_SCENERY, GameAc
 private:
     TileElement* FindSceneryElement() const
     {
-        TileElement* tileElement = map_get_first_element_at(_x / 32, _y / 32);
+        TileElement* tileElement = map_get_first_element_at(_loc.x >> 5, _loc.y >> 5);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/SmallSceneryRemoveAction.hpp]: /32

File: src/openrct2/actions/SmallSceneryRemoveAction.hpp [hunk 26]
```diff
@@ -153,7 +145,7 @@ DEFINE_GAME_ACTION(SmallSceneryRemoveAction, GAME_COMMAND_REMOVE_SCENERY, GameAc
                 continue;
             if ((tileElement->AsSmallScenery()->GetSceneryQuadrant()) != _quadrant)
                 continue;
-            if (tileElement->base_height != _baseHeight)
+            if (tileElement->base_height != _loc.z >> 3)
```
Comment (duncanspumpkin) [inline src/openrct2/actions/SmallSceneryRemoveAction.hpp]: /8

File: src/openrct2/actions/WallRemoveAction.hpp [hunk 27]
```diff
@@ -76,37 +76,37 @@ DEFINE_GAME_ACTION(WallRemoveAction, GAME_COMMAND_REMOVE_WALL, GameActionResult)
...
-        res->Position.x = (_location.x << 5) + 16;
-        res->Position.y = (_location.y << 5) + 16;
+        res->Position.x = _loc.x + 16;
+        res->Position.y = _loc.y + 16;
         res->Position.z = tile_element_height(res->Position.x, res->Position.y);
 
         tile_element_remove_banner_entry(wallElement);
         map_invalidate_tile_zoom1(
-            _location.x << 5, _location.y << 5, wallElement->base_height * 8, (wallElement->base_height * 8) + 72);
+            _loc.x, _loc.y , wallElement->base_height << 3, (wallElement->base_height << 3)  + 72);
```
Comment (duncanspumpkin) [inline src/openrct2/actions/WallRemoveAction.hpp]: *8

File: src/openrct2/actions/WallRemoveAction.hpp [hunk 28]
```diff
@@ -76,37 +76,37 @@ DEFINE_GAME_ACTION(WallRemoveAction, GAME_COMMAND_REMOVE_WALL, GameActionResult)
...
 
         return res;
     }
 
 private:
-    TileElement* GetFirstWallElementAt(const TileCoordsXYZD& location, bool isGhost) const
+    TileElement* GetFirstWallElementAt(const CoordsXYZD& location, bool isGhost) const
     {
-        TileElement* tileElement = map_get_first_element_at(location.x, location.y);
+        TileElement* tileElement = map_get_first_element_at(location.x >> 5 , location.y >> 5 );
```
Comment (duncanspumpkin) [inline src/openrct2/actions/WallRemoveAction.hpp]: /32

File: src/openrct2/actions/WallRemoveAction.hpp [hunk 29]
```diff
@@ -76,37 +76,37 @@ DEFINE_GAME_ACTION(WallRemoveAction, GAME_COMMAND_REMOVE_WALL, GameActionResult)
...
         if (!tileElement)
             return nullptr;
 
         do
         {
             if (tileElement->GetType() != TILE_ELEMENT_TYPE_WALL)
                 continue;
-            if (tileElement->base_height != location.z)
+            if (tileElement->base_height != location.z >> 3)
```
Comment (duncanspumpkin) [inline src/openrct2/actions/WallRemoveAction.hpp]: /8

File: src/openrct2/ride/TrackDesign.cpp [hunk 30]
```diff
@@ -776,17 +776,17 @@ static bool TrackDesignPlaceSceneryElementRemoveGhost(
                 quadrant = 0;
             }
 
-            ga = std::make_unique<SmallSceneryRemoveAction>(mapCoord.x, mapCoord.y, z, quadrant, entry_index);
+            ga = std::make_unique<SmallSceneryRemoveAction>(CoordsXYZ{ mapCoord.x, mapCoord.y, z << 3 }, quadrant, entry_index);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackDesign.cpp]: *8

File: src/openrct2/ride/TrackDesign.cpp [hunk 31]
```diff
@@ -776,17 +776,17 @@ static bool TrackDesignPlaceSceneryElementRemoveGhost(
...
             break;
         }
         case OBJECT_TYPE_LARGE_SCENERY:
-            ga = std::make_unique<LargeSceneryRemoveAction>(mapCoord.x, mapCoord.y, z, sceneryRotation, 0);
+            ga = std::make_unique<LargeSceneryRemoveAction>(CoordsXYZD{ mapCoord.x, mapCoord.y, z << 3, sceneryRotation }, 0);
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackDesign.cpp]: *8

File: src/openrct2/ride/TrackDesign.cpp [hunk 32]
```diff
@@ -776,17 +776,17 @@ static bool TrackDesignPlaceSceneryElementRemoveGhost(
...
             break;
         case OBJECT_TYPE_WALLS:
-            ga = std::make_unique<WallRemoveAction>(TileCoordsXYZD{ mapCoord.x / 32, mapCoord.y / 32, z, sceneryRotation });
+            ga = std::make_unique<WallRemoveAction>(CoordsXYZD{ mapCoord.x, mapCoord.y , z << 3, sceneryRotation });
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackDesign.cpp]: *8

File: src/openrct2/ride/TrackDesign.cpp [hunk 33]
```diff
@@ -776,17 +776,17 @@ static bool TrackDesignPlaceSceneryElementRemoveGhost(
...
             break;
         case OBJECT_TYPE_PATHS:
-            ga = std::make_unique<FootpathRemoveAction>(mapCoord.x, mapCoord.y, z);
+            ga = std::make_unique<FootpathRemoveAction>(CoordsXYZ{ mapCoord.x, mapCoord.y, z << 3 });
```
Comment (duncanspumpkin) [inline src/openrct2/ride/TrackDesign.cpp]: *8

File: src/openrct2/world/Footpath.cpp [hunk 34]
```diff
@@ -127,7 +127,7 @@ TileElement* map_get_footpath_element(int32_t x, int32_t y, int32_t z)
 
 money32 footpath_remove(int32_t x, int32_t y, int32_t z, int32_t flags)
 {
-    auto action = FootpathRemoveAction(x, y, z);
+    auto action = FootpathRemoveAction({ x, y, z << 3 });
```
Comment (duncanspumpkin) [inline src/openrct2/world/Footpath.cpp]: *8

File: src/openrct2/world/Map.cpp [hunk 35]
```diff
@@ -1756,15 +1756,15 @@ static void clear_element_at(int32_t x, int32_t y, TileElement** elementPtr)
         }
         case TILE_ELEMENT_TYPE_WALL:
         {
-            TileCoordsXYZD wallLocation = { x >> 5, y >> 5, element->base_height, element->GetDirection() };
+            CoordsXYZD wallLocation = { x, y, element->base_height << 3, element->GetDirection() };
```
Comment (duncanspumpkin) [inline src/openrct2/world/Map.cpp]: *8

File: src/openrct2/world/Map.cpp [hunk 36]
```diff
@@ -1756,15 +1756,15 @@ static void clear_element_at(int32_t x, int32_t y, TileElement** elementPtr)
...
             auto wallRemoveAction = WallRemoveAction(wallLocation);
             GameActions::Execute(&wallRemoveAction);
         }
         break;
         case TILE_ELEMENT_TYPE_LARGE_SCENERY:
         {
             auto removeSceneryAction = LargeSceneryRemoveAction(
-                x, y, element->base_height, element->GetDirection(), element->AsLargeScenery()->GetSequenceIndex());
+                { x, y, element->base_height << 3, element->GetDirection()}, element->AsLargeScenery()->GetSequenceIndex());
```
Comment (duncanspumpkin) [inline src/openrct2/world/Map.cpp]: *8

File: src/openrct2/world/Scenery.cpp [hunk 37]
```diff
@@ -186,7 +186,7 @@ void scenery_remove_ghost_tool_placement()
     {
         gSceneryGhostType &= ~SCENERY_GHOST_FLAG_0;
 
-        auto removeSceneryAction = SmallSceneryRemoveAction(x, y, z, gSceneryQuadrant, gSceneryPlaceObject);
+        auto removeSceneryAction = SmallSceneryRemoveAction({ x, y, z << 3 }, gSceneryQuadrant, gSceneryPlaceObject);
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.cpp]: *8

File: src/openrct2/world/Scenery.cpp [hunk 38]
```diff
@@ -195,7 +195,7 @@ void scenery_remove_ghost_tool_placement()
     if (gSceneryGhostType & SCENERY_GHOST_FLAG_1)
     {
         gSceneryGhostType &= ~SCENERY_GHOST_FLAG_1;
-        TileElement* tileElement = map_get_first_element_at(x / 32, y / 32);
+        TileElement* tileElement = map_get_first_element_at(x >> 5, y >> 5);
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.cpp]: /32

File: src/openrct2/world/Scenery.cpp [hunk 39]
```diff
@@ -205,7 +205,7 @@ void scenery_remove_ghost_tool_placement()
             if (tileElement->base_height != z)
                 continue;
 
-            auto footpathSceneryRemoveAction = FootpathSceneryRemoveAction({ x, y, z * 8 });
+            auto footpathSceneryRemoveAction = FootpathSceneryRemoveAction({ x, y, z << 3 });
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.cpp]: *8

File: src/openrct2/world/Scenery.cpp [hunk 40]
```diff
@@ -215,8 +215,8 @@ void scenery_remove_ghost_tool_placement()
     if (gSceneryGhostType & SCENERY_GHOST_FLAG_2)
     {
         gSceneryGhostType &= ~SCENERY_GHOST_FLAG_2;
-
-        TileCoordsXYZD wallLocation = { x >> 5, y >> 5, z, gSceneryGhostWallRotation };
+        
+        CoordsXYZD wallLocation = { x , y, z << 3, gSceneryGhostWallRotation };
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.cpp]: *8

File: src/openrct2/world/Scenery.cpp [hunk 41]
```diff
@@ -226,7 +226,7 @@ void scenery_remove_ghost_tool_placement()
     {
         gSceneryGhostType &= ~SCENERY_GHOST_FLAG_3;
 
-        auto removeSceneryAction = LargeSceneryRemoveAction(x, y, z, gSceneryPlaceRotation, 0);
+        auto removeSceneryAction = LargeSceneryRemoveAction({ x, y, z << 3, gSceneryPlaceRotation }, 0);
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.cpp]: *8

File: src/openrct2/world/Scenery.cpp [hunk 42]
```diff
@@ -237,7 +237,7 @@ void scenery_remove_ghost_tool_placement()
     {
         gSceneryGhostType &= ~SCENERY_GHOST_FLAG_4;
 
-        auto removeSceneryAction = BannerRemoveAction({ x, y, z * 8, gSceneryPlaceRotation });
+        auto removeSceneryAction = BannerRemoveAction({ x, y, z << 3, gSceneryPlaceRotation });
```
Comment (duncanspumpkin) [inline src/openrct2/world/Scenery.cpp]: *8

**Comment by 0Cufox0** [comment]
I removed all the shifting!

**Comment by duncanspumpkin** [comment]
Okay that's good. There is a merge conflict you will need to correct. (Do a rebase against develop and force push after correcting it). It's a bit annoying the replay issue but that is something that needs correcting outside this pr. Because of the replay issue this can't be merged just yet. To fix the replay issue we really need to make a replay repo and change some build steps.

**Comment by 0Cufox0** [comment]
> Okay that's good. There is a merge conflict you will need to correct. (Do a rebase against develop and force push after correcting it). It's a bit annoying the replay issue but that is something that needs correcting outside this pr. Because of the replay issue this can't be merged just yet. To fix the replay issue we really need to make a replay repo and change some build steps.

I don't get it, I think I broke something :<

**Comment by duncanspumpkin** [comment]
I think what is happening is you haven't updated your develop branch so when you rebase it is not doing anything. First update your develop to match this repo then rebase this branch. If you can't get it working just wait and i can fix it. 

On the replay side. I tried to start moving the replays yesterday. Encountered a bit of an issue but should get it sorted soon. Hopefully be ready in the next few days.

**Comment by 0Cufox0** [comment]
Yeah I'm too scared to push the stuff I did(it says I have 47 commits on this branch), I guess I'll wait!

File: src/openrct2-android/app/build.gradle [hunk 43]
```diff
@@ -98,7 +98,7 @@ android.applicationVariants.all { variant ->
             into "$variant.mergeAssets.outputDir/data/title"
         }
         download {
-            src 'https://github.com/OpenRCT2/objects/releases/download/v1.0.11/objects.zip'
+            src 'https://github.com/OpenRCT2/objects/releases/download/v1.0.10/objects.zip'
```
Comment (duncanspumpkin) [inline src/openrct2-android/app/build.gradle]: undo this change.

File: src/openrct2/network/Network.cpp [hunk 44]
```diff
@@ -34,7 +34,8 @@
 // This string specifies which version of network stream current build uses.
 // It is used for making sure only compatible builds get connected, even within
 // single OpenRCT2 version.
-#define NETWORK_STREAM_VERSION "0"
+
+#define NETWORK_STREAM_VERSION "47"
```
Comment (duncanspumpkin) [inline src/openrct2/network/Network.cpp]: should be version 2 now

File: src/openrct2-android/app/build.gradle [hunk 45]
```diff
@@ -98,7 +98,7 @@ android.applicationVariants.all { variant ->
             into "$variant.mergeAssets.outputDir/data/title"
         }
         download {
-            src 'https://github.com/OpenRCT2/objects/releases/download/v1.0.10/objects.zip'
+            src 'https://github.com/OpenRCT2/objects/releases/download/v1.0.11/objects.zip'
```
Comment (AaronVanGeffen) [inline src/openrct2-android/app/build.gradle]: Why is this in this commit? (51315aa0444ccab58a229a7570ee93f8e69caf1f — weird commit message, too.)

Comment (duncanspumpkin) [reply src/openrct2-android/app/build.gradle]: Because of a mistake in the second commit. I meant to fix it in the last rebase but forgot about this change.

Comment (duncanspumpkin) [reply src/openrct2-android/app/build.gradle]: Fixed now
